cpp/src/arrow/compute/kernels/scalar_set_lookup_test.cc

```diff
@@ -238,6 +239,29 @@ TEST_F(TestIsInKernel, TimeTimestamp) {
             "[true, false, true]");
 }
 
+TEST_F(TestIsInKernel, TimeDuration) {
+  for (const auto& type : DurationTypes()) {
+    CheckIsIn(type, "[1, null, 5, 1, 2]", "[2, 1, null]",
+              "[true, true, false, true, true]", /*skip_nulls=*/false);
+    CheckIsIn(type, "[1, null, 5, 1, 2]", "[2, 1, null]",
+              "[true, false, false, true, true]", /*skip_nulls=*/true);
+
+    // Duplicates in right array
+    CheckIsIn(type, "[1, null, 5, 1, 2]", "[2, 1, 1, null, 2]",
+              "[true, true, false, true, true]", /*skip_nulls=*/false);
+    CheckIsIn(type, "[1, null, 5, 1, 2]", "[2, 1, 1, null, 2]",
+              "[true, false, false, true, true]", /*skip_nulls=*/true);
+  }
+
+  // Different units, invalid cast
```

Comment:
Interesting... ideally, the values could be cast to MILLI instead of casting the value_set to SECOND. Don't have to solve this here, though. :-)

Reply:
#36204 will solve this.